# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/components/newtab/test/unit/content-src/components/DiscoveryStreamComponents/DSCard.test.jsx
# Commit: 626dd2a045af
# Full Hash: 626dd2a045af6cd510ef04e67d027909509547f1
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2020-04-17 21:46:50
# Regressor Bug: 1614502
# File Overlap Count: 2
# Description:
#   Bug 1614502 - Add a mechanism that can generate an about:home document for a startup cache. r=Gijs,k88hudson
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D68094
# ==============================================================================

diff -r 7ec776d31a2c -r 626dd2a045af browser/components/newtab/test/unit/content-src/components/DiscoveryStreamComponents/DSCard.test.jsx
--- a/browser/components/newtab/test/unit/content-src/components/DiscoveryStreamComponents/DSCard.test.jsx	Fri Apr 17 15:31:11 2020 +0000
+++ b/browser/components/newtab/test/unit/content-src/components/DiscoveryStreamComponents/DSCard.test.jsx	Fri Apr 17 15:31:39 2020 +0000
@@ -1,5 +1,5 @@
 import {
-  DSCard,
+  _DSCard as DSCard,
   DefaultMeta,
   PlaceholderDSCard,
   CTAButtonMeta,
@@ -15,12 +15,18 @@
 import { shallow, mount } from "enzyme";
 import { FluentOrText } from "content-src/components/FluentOrText/FluentOrText";
 
+const DEFAULT_PROPS = {
+  App: {
+    isForStartupCache: false,
+  },
+};
+
 describe("<DSCard>", () => {
   let wrapper;
   let sandbox;
 
   beforeEach(() => {
-    wrapper = shallow(<DSCard />);
+    wrapper = shallow(<DSCard {...DEFAULT_PROPS} />);
     wrapper.setState({ isSeen: true });
     sandbox = sinon.createSandbox();
   });
@@ -80,7 +86,7 @@
   });
 
   it("should render badges for pocket, bookmark when not a spoc element ", () => {
-    wrapper = mount(<DSCard context_type="bookmark" />);
+    wrapper = mount(<DSCard context_type="bookmark" {...DEFAULT_PROPS} />);
     wrapper.setState({ isSeen: true });
     const contextFooter = wrapper.find(DSContextFooter);
 
@@ -89,7 +95,9 @@
 
   it("should render Sponsored Context for a spoc element", () => {
     const context = "Sponsored by Foo";
-    wrapper = mount(<DSCard context_type="bookmark" context={context} />);
+    wrapper = mount(
+      <DSCard context_type="bookmark" context={context} {...DEFAULT_PROPS} />
+    );
     wrapper.setState({ isSeen: true });
     const contextFooter = wrapper.find(DSContextFooter);
 
@@ -102,7 +110,7 @@
 
     beforeEach(() => {
       dispatch = sandbox.stub();
-      wrapper = shallow(<DSCard dispatch={dispatch} />);
+      wrapper = shallow(<DSCard dispatch={dispatch} {...DEFAULT_PROPS} />);
       wrapper.setState({ isSeen: true });
     });
 
@@ -191,7 +199,7 @@
 
   describe("DSCard with CTA", () => {
     beforeEach(() => {
-      wrapper = mount(<DSCard />);
+      wrapper = mount(<DSCard {...DEFAULT_PROPS} />);
       wrapper.setState({ isSeen: true });
     });
 
@@ -265,7 +273,7 @@
   });
   describe("DSCard with Intersection Observer", () => {
     beforeEach(() => {
-      wrapper = shallow(<DSCard />);
+      wrapper = shallow(<DSCard {...DEFAULT_PROPS} />);
     });
 
     it("should render card when seen", () => {
@@ -297,7 +305,7 @@
     });
 
     it("should setup observer on componentDidMount", () => {
-      wrapper = mount(<DSCard />);
+      wrapper = mount(<DSCard {...DEFAULT_PROPS} />);
       assert.isTrue(!!wrapper.instance().observer);
     });
   });
@@ -307,7 +315,7 @@
       cancelIdleCallback: sinon.stub(),
     };
     beforeEach(() => {
-      wrapper = shallow(<DSCard windowObj={windowStub} />);
+      wrapper = shallow(<DSCard windowObj={windowStub} {...DEFAULT_PROPS} />);
     });
 
     it("should call requestIdleCallback on componentDidMount", () => {
@@ -319,34 +327,45 @@
       assert.calledOnce(windowStub.cancelIdleCallback);
     });
   });
+  describe("DSCard when rendered for about:home startup cache", () => {
+    beforeEach(() => {
+      const props = {
+        App: {
+          isForStartupCache: true,
+        },
+      };
+      wrapper = mount(<DSCard {...props} />);
+    });
+
+    it("should be set as isSeen automatically", () => {
+      assert.isTrue(wrapper.instance().state.isSeen);
+    });
+  });
 });
 
 describe("<PlaceholderDSCard> component", () => {
   it("should have placeholder prop", () => {
     const wrapper = shallow(<PlaceholderDSCard />);
-    const card = wrapper.find(DSCard);
-    assert.lengthOf(card, 1);
-
-    const placeholder = wrapper.find(DSCard).prop("placeholder");
+    const placeholder = wrapper.prop("placeholder");
     assert.isTrue(placeholder);
   });
 
   it("should contain placeholder div", () => {
-    const wrapper = shallow(<DSCard placeholder={true} />);
+    const wrapper = shallow(<DSCard placeholder={true} {...DEFAULT_PROPS} />);
     wrapper.setState({ isSeen: true });
     const card = wrapper.find("div.ds-card.placeholder");
     assert.lengthOf(card, 1);
   });
 
   it("should not be clickable", () => {
-    const wrapper = shallow(<DSCard placeholder={true} />);
+    const wrapper = shallow(<DSCard placeholder={true} {...DEFAULT_PROPS} />);
     wrapper.setState({ isSeen: true });
     const anchor = wrapper.find("SafeAnchor.ds-card-link");
     assert.lengthOf(anchor, 0);
   });
 
   it("should not have context menu", () => {
-    const wrapper = shallow(<DSCard placeholder={true} />);
+    const wrapper = shallow(<DSCard placeholder={true} {...DEFAULT_PROPS} />);
     wrapper.setState({ isSeen: true });
     const linkMenu = wrapper.find(DSLinkMenu);
     assert.lengthOf(linkMenu, 0);
